src/imp/parser.rs

```diff
@@ -124,7 +124,13 @@ impl<'a> PyParser<'a> {
         }
         let head = self.parse_expr(false)?;
         self.skip_trivia();
-        if self.starts_with(":") { self.parse_map_init(head)? } else { self.parse_sup(head)? }
+        if self.try_consume(",") {
+          self.parse_sup(head)?
+        } else if self.try_consume(":") {
+          self.parse_map_init(head)?
+        } else {
+          self.expected("Comma or Colon")?
```

Comment:
```suggestion
          self.expected("',' or ':")?
```
To be consistent with how we're displaying this in other situations